pkg/cstorpool/v1alpha2/cstorpool.go

```diff
@@ -90,9 +94,30 @@ func (b *listBuilder) WithUIDs(poolUIDs ...string) *listBuilder {
 	return b
 }
 
+// WithObjects builds a list of cstor pools
+// based on the provided pool
+func (b *listBuilder) WithObjects(pools ...*apis.CStorPool) *listBuilder {
+	for _, pool := range pools {
+		item := &csp{object: pool}
+		b.list.items = append(b.list.items, item)
+	}
+	return b
+}
+
 // List returns the list of csp
 // instances that were built by
 // this builder
 func (b *listBuilder) List() *cspList {
 	return b.list
 }
+
+// ListNodeName returns the list of nodenames
+// of csp that were built by
+// this builder
+func (b *listBuilder) ListNodeNames() []string {
```

Comment:
Create another function called `ListUniqueNodeNames` and make use of it in test code

Reply:
done